commit/chainfee/observation.go

```diff
@@ -28,10 +30,36 @@ func (p *processor) Observation(
 ) (Observation, error) {
 	lggr := logutil.WithContextValues(ctx, p.lggr)
 
-	feeComponents := p.obs.getChainsFeeComponents(ctx, lggr)
-	nativeTokenPrices := p.obs.getNativeTokenPrices(ctx, lggr)
-	chainFeeUpdates := p.obs.getChainFeePriceUpdates(ctx, lggr)
-	fChain := p.observeFChain(lggr)
+	var (
+		feeComponents     map[cciptypes.ChainSelector]types.ChainFeeComponents
+		nativeTokenPrices map[cciptypes.ChainSelector]cciptypes.BigInt
+		chainFeeUpdates   map[cciptypes.ChainSelector]Update
+		fChain            map[cciptypes.ChainSelector]int
+	)
+
+	operations := []func(ctx context.Context, l logger.Logger){
+		func(ctx context.Context, l logger.Logger) { feeComponents = p.obs.getChainsFeeComponents(ctx, l) },
+		func(ctx context.Context, l logger.Logger) { nativeTokenPrices = p.obs.getNativeTokenPrices(ctx, l) },
+		func(ctx context.Context, l logger.Logger) { chainFeeUpdates = p.obs.getChainFeePriceUpdates(ctx, l) },
+		func(ctx context.Context, l logger.Logger) { fChain = p.observeFChain(l) },
+	}
+
+	callTimeout := p.cfg.ChainFeeAsyncObserverSyncTimeout
+	callCtx, cf := context.WithTimeout(ctx, callTimeout)
+	defer cf()
+	lggr.Debugw("spawning observing goroutines", "timeout", callTimeout)
+
+	wg := sync.WaitGroup{}
+	wg.Add(len(operations))
+	for i, op := range operations {
+		go func(i int, op func(ctx context.Context, l logger.Logger)) {
+			defer wg.Done()
+			tStart := time.Now()
+			op(callCtx, logger.With(lggr, "opID", i))
+			lggr.Debugw("observing goroutine finished", "opID", i, "duration", time.Since(tStart))
```

Comment:
This opID doesn't bring much value, I would consider passing some meaningful/human understandable value here via operations lambda or sth

for instance:
```go
operations := map[string]func(context, logger){
   "feecomponents": func(ctx context.Context, l logger.Logger) { feeComponents = p.obs.getChainsFeeComponents(ctx, l) }
   "nativeTokenPrices": ...
}
// and then


for opName, op := range operations {
			lggr.Debugw("observing goroutine finished", "operation", opName, "duration", time.Since(tStart))
}
```